lib/std/setutils.nim

```diff
@@ -0,0 +1,33 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2020 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This module adds functionality to the default set
+
+import typetraits
+
+type SetElement* = char|byte|bool|int16|uint16|enum|uint8|int8
+
+template toSet*(iter: untyped): untyped =
+  ##Iterates through an openArray making a set from it.
```

Comment:
spacing + other fixes:
```
## Return a builtin set from the elements of iterable `iter`
```
(note: later versions of nim might allow: `proc toSet*[T](iter: Iterable[T]): set[T] =`)